crates/matrix-sdk/src/account.rs

```diff
@@ -133,9 +133,30 @@ impl Account {
         let config = Some(RequestConfig::new().force_auth());
 
         let response = self.client.send(request, config).await?;
+        if let Some(url) = response.avatar_url.clone() {
+            // If an avatar is found cache it.
+            let url_data = rmp_serde::to_vec(&url.as_str())?;
+            self.client.store().set_custom_value(user_id.as_bytes(), url_data).await?;
+        } else {
+            // If there is no avatar the user has removed it and we uncache it.
+            self.client.store().remove_custom_value(user_id.as_bytes()).await?;
```

Comment:
I was wondering, should I avoid using the `?` here? Maybe we do not want this API call to fail, if the caching fails.
Should I just consider the Ok cases and do nothing in the Error cases?